analytics/src/main/java/org/odk/collect/analytics/Analytics.kt

```diff
@@ -14,4 +14,18 @@ interface Analytics {
     fun logNonFatal(message: String)
     fun setAnalyticsCollectionEnabled(isAnalyticsEnabled: Boolean)
     fun setUserProperty(name: String, value: String)
+
+    companion object {
+
+        private var instance: Analytics = NoopAnalytics()
+
+        fun setInstance(analytics: Analytics) {
+            this.instance = analytics
+        }
+
+        @JvmStatic
+        fun log(event: String) {
```

Comment:
This is potentially a controversial change. Because we never test analytics (mainly because we often delete calls soon after adding them, and they are never critical) and because we never depend on its behaviour, it seems OK to me to have a static method for logging events. This of course goes against everything I would usually say, but I'll point you back to the last sentence in terms of justification for this rule break 😆.

For me, the advantage of this is that we don't have to rework constructor/method signatures when we add remove analytics calls from a component.

Reply:
Absolutely, I think this is pragmatic given we’ve decide we will very rarely if ever test analytics.

Reply:
I want to say it more strongly -- I think this is a good decision because it makes adding and removing analytics very lightweight.